### ISSUE TITLE
[ISSUE] MySQL migration produce statements in the wrong order

### ISSUE THREAD

**Issue by JesusTheHun** [drizzle-team/drizzle-kit-mirror#45]
Running migration for the following schemas :
```ts
export const authorsTable = mysqlTable('authors', {
  id: int('id').autoincrement().primaryKey(),
  nickname: varchar('nickname', { length: 255 }).notNull(),
}, (table) => ({
  nickname: index('nickname').on(table.nickname),
}));

export const booksTable = mysqlTable('books', {
  id: int('id').autoincrement().primaryKey(),
  authorNickname: int('authorNickname').notNull().references(() => authorsTable.nickname),
});
```

Will produce the following SQL script :

```sql
CREATE TABLE `authors` (
	`id` int AUTO_INCREMENT PRIMARY KEY NOT NULL,
	`nickname` varchar(255) NOT NULL
);

CREATE TABLE `books` (
	`id` int AUTO_INCREMENT PRIMARY KEY NOT NULL,
	`authorNickname` int NOT NULL
);

ALTER TABLE `books` ADD CONSTRAINT `books_authorNickname_authors_nickname_fk` FOREIGN KEY (`authorNickname`) REFERENCES `authors`(`nickname`);
CREATE INDEX `nickname` ON `authors` (`nickname`);
```

This will fail because the **index** should be created **before** adding the **constraint**.

**Comment by AndriiSherman** [comment]
agree, changing it

**Comment by AndriiSherman** [comment]
Fixed in `drizzle-kit@0.18.1`
